### PR TITLE
Okd 4.3

### PR DESCRIPTION
**Pull request by qinqon**

### CODE REVIEW

File: cluster-provision/okd/4.3/provision.sh [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+
+${gocli} provision okd \
+--prefix okd-4.3-provision \
+--dir-scripts ${PARENT_DIR}/okd/scripts \
+--dir-manifests ${PARENT_DIR}/manifests \
+--dir-hacks ${PARENT_DIR}/okd/hacks \
+--master-memory 10240 \
+--installer-pull-token-file ${INSTALLER_PULL_SECRET} \
+--installer-repo-tag release-4.3 \
+--installer-release-image registry.svc.ci.openshift.org/ocp/release@sha256:77dbb7cb39bc69e5de65bf50a87942de6a98bc4dc2a2d0a9d539db004a36d964 \
```
Comment (cynepco3hahue) [inline cluster-provision/okd/4.3/provision.sh]: can you remove usage of the internal image, I think `--installer-repo-tag release-4.3` should be enough.

File: cluster-provision/README.md [hunk 2]
```diff
@@ -32,11 +32,13 @@
 
 * `okd-base` contains all needed packages to provision and run OKD cluster on top of the libvirt provider
 * `okd-4.1` okd-4.1 cluster provisioned with OpenShift installer on top of the libvirt provider, this image contains custom libvirt image that includes fixes to deploy new nodes without need to apply any W/A
+* `okd-4.2` okd-4.2 cluster provisioned with OpenShift installer on top of the libvirt provider, this image contains custom libvirt image that includes fixes to deploy new nodes without need to apply any W/A
```
Comment (cynepco3hahue) [inline cluster-provision/README.md]: Add new provider under the README

File: cluster-provision/okd/scripts/provision.sh [hunk 3]
```diff
@@ -13,6 +13,15 @@ compile_installer () {
     local build_pkgs="git golang-bin gcc-c++"
     dnf install -y ${build_pkgs}
 
+    # install golang
+    go_version=1.12.12
+    curl https://dl.google.com/go/go${go_version}.linux-amd64.tar.gz -o go.tar.gz
+    tar -xvzf go.tar.gz -C /usr/local/
+    rm go.tar.gz
```
Comment (cynepco3hahue) [inline cluster-provision/okd/scripts/provision.sh]: rm -rf

File: cluster-provision/okd/4.3/provision.sh (hunk 1 above)
Comment (qinqon) [reply cluster-provision/okd/4.3/provision.sh]: Sure, forgot to remove it.

**Comment by qinqon** [comment]
@cynepco3hahue this is draft github PR, let's see how it goes, btw how do I create the prow jobs for okd-4.3 provision ?

**Comment by cynepco3hahue** [comment]
You can check the similar okd-4,2 job here - https://github.com/kubevirt/project-infra/blob/master/github/ci/prow/files/jobs/kubevirtci/kubevirtci-presubmits.yaml

**Comment by qinqon** [comment]
@cynepco3hahue https://github.com/kubevirt/project-infra/pull/237

**Comment by qinqon** [comment]
Previous timeout was transitory, real issue is deploying CNAO 

      message: |-
        DaemonSet "linux-bridge/kube-cni-linux-bridge-plugin" is not available (awaiting 1 nodes)
        Deployment "kubemacpool-system/kubemacpool-mac-controller-manager" is not available (awaiting 1 nodes)

**Comment by qinqon** [comment]
One of the three pod expected for linux-bridge has not enough cpu and memory

status:
  conditions:
  - lastProbeTime: null
    lastTransitionTime: "2019-10-30T16:48:38Z"
    message: '0/3 nodes are available: 2 Insufficient cpu, 2 Insufficient memory,
      2 node(s) didn''t match node selector.'
    reason: Unschedulable
    status: "False"
    type: PodScheduled
  phase: Pending
  qosClass: Guaranteed

**Comment by qinqon** [comment]
Ok memory issue is gone, but we still need two cpus 

 conditions:
  - lastProbeTime: null
    lastTransitionTime: "2019-11-04T12:52:30Z"
    message: '0/3 nodes are available: 2 Insufficient cpu, 2 node(s) didn''t match
      node selector.'
    reason: Unschedulable
    status: "False"
    type: PodScheduled
  phase: Pending
  qosClass: Guaranteed

**Comment by qinqon** [comment]
2 vcpus parameter is correctly set 
```
[root@b65c7a3bd76b /]# oc get machineset --all-namespaces -o yaml  |grep Vcpu
            domainVcpu: 2
```

but the workers has only one core

**Comment by qinqon** [comment]
libvirt is correctly set

```xml
[root@b65c7a3bd76b /]# virsh dumpxml test-1-rjvzz-worker-0-z6wg2 |grep cpu
  <vcpu placement='static'>2</vcpu>
  <cpu mode='host-passthrough' check='none'/>
```

**Comment by qinqon** [comment]
We have pass CNAO issue, now looks like we have some roles issue

        error syncing local storage: error applying pv cluster role binding local-storage-provisioner-pv-binding: clusterrolebindings.rbac.authorization.k8s.io "local-storage-provisioner-pv-binding" is forbidden: user "system:serviceaccount:local-storage:local-storage-operator" (groups=["system:serviceaccounts" "system:serviceaccounts:local-storage" "system:authenticated"]) is attempting to grant RBAC permissions not currently held:
        {APIGroups:["events.k8s.io"], Resources:["events"], Verbs:["create" "patch" "update"]}
      status: "False"
      type: Available
    readyReplicas: 0

**Comment by qinqon** [comment]
Looks like this 
https://bugzilla.redhat.com/show_bug.cgi?id=1743482

**Comment by qinqon** [comment]
Let's re-run check-provision, last one has fail checking nodes https://prow.apps.ovirt.org/view/gcs/kubevirt-prow/pr-logs/pull/kubevirt_kubevirtci/170/check-provision-okd-4.3/1191739444252643328

**Comment by qinqon** [comment]
check-provision-okd-4.3

File: cluster-provision/README.md [hunk 4]
```diff
@@ -32,11 +32,14 @@
 
 * `okd-base` contains all needed packages to provision and run OKD cluster on top of the libvirt provider
 * `okd-4.1` okd-4.1 cluster provisioned with OpenShift installer on top of the libvirt provider, this image contains custom libvirt image that includes fixes to deploy new nodes without need to apply any W/A
+* `okd-4.3` okd-4.3 cluster provisioned with OpenShift installer on top of the libvirt provider, this image contains custom libvirt image that includes fixes to deploy new nodes without need to apply any W/A
 
 ## Versions to use
 
 * `kubevirtci/okd-base`: `sha256:259e776998da3a503a30fdf935b29102443b24ca4ea095c9478c37e994e242bb`
 * `kubevirtci/okd-4.1`: `sha256:b26decaf0454cc1f99b39b9bc991f715a8d6d8e97499db5a2d40f778430972f7`
+* `kubevirtci/okd-4.1`: `sha256:2b7b5e09b9bdf2ca40b8e153a111702584e2a3e802643e3e7df1f2d97eca0ce8`
```
Comment (cynepco3hahue) [inline cluster-provision/README.md]: remove duplication

File: cluster-provision/okd/4.3/provision.sh [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+--dir-scripts ${PARENT_DIR}/okd/scripts \
+--dir-manifests ${PARENT_DIR}/manifests \
+--dir-hacks ${PARENT_DIR}/okd/hacks \
+--workers-memory 8192 \
+--workers-cpu 4 \
+--installer-pull-token-file ${INSTALLER_PULL_SECRET} \
+--installer-repo-tag release-4.3 \
+--installer-release-image registry.svc.ci.openshift.org/origin/release:4.3 \
+"kubevirtci/okd-base@${okd_base_hash}"
+rc=$?
```
Comment (cynepco3hahue) [inline cluster-provision/okd/4.3/provision.sh]: will `exit $?` work?

File: cluster-provision/okd/scripts/run.sh [hunk 6]
```diff
@@ -70,6 +79,8 @@ done
 # wait half minute, just to be sure that we do not get old cluster state
 sleep 30
 
+uncordon_nodes
```
Comment (cynepco3hahue) [inline cluster-provision/okd/scripts/run.sh]: hm do we still need to uncordon workers, also with 2 workers on install time?

Comment (qinqon) [reply cluster-provision/okd/scripts/run.sh]: That's right, also I am going to wait at scale down at provision.

File: cluster-provision/okd/4.3/provision.sh (hunk 5 above)
Comment (qinqon) [reply cluster-provision/okd/4.3/provision.sh]: Yep, I have move the docker kill up, does not make sense to store rc and exit.

**Comment by qinqon** [comment]
check-provision-okd-4.3

File: cluster-provision/okd/scripts/run.sh (hunk 6 above)
Comment (cynepco3hahue) [reply cluster-provision/okd/scripts/run.sh]: I still can see uncordon methods under the run, can you please re-check?

**Pull request merged**